tools/replay-archive/src/cli.rs

```diff
@@ -318,23 +334,34 @@ impl ReplayController {
             shard_context,
         )?;
 
-        let apply_result = match shard_update_result {
-            ShardUpdateResult::NewChunk(result) => result.apply_result,
-            ShardUpdateResult::OldChunk(result) => result.apply_result,
+        let output = match shard_update_result {
+            ShardUpdateResult::NewChunk(NewChunkResult {
+                gas_limit,
+                shard_uid: _,
+                apply_result,
+                resharding_results: _,
+            }) => {
+                let chunk_extra = resulting_chunk_extra(&apply_result, gas_limit, protocol_version);
+                ReplayChunkOutput { chunk_extra }
+            }
+            ShardUpdateResult::OldChunk(OldChunkResult {
+                shard_uid: _,
+                apply_result,
+                resharding_results: _,
+            }) => {
+                let mut chunk_extra = ChunkExtra::clone(&prev_chunk_extra.as_ref());
+                *chunk_extra.state_root_mut() = apply_result.new_root;
+                ReplayChunkOutput { chunk_extra }
+            }
             ShardUpdateResult::Resharding(_) => bail!("Unexpected apply result for resharding"),
         };
 
-        // TODO: This uses ChunkExtra to check the result of apply-chunk.
-        // Make this check independent of the existing ChunkExtra (so no need to store it in the DB).
-        check_apply_block_result(
```

Comment:
no need to call this anymore since `validate_chunk_with_chunk_extra` (run inside `validate_chunk`) does the same checks comparing the information in the previously recorded ChunkExtra with the information in the next chunks.